File: js/linkedlist.js
 class Node {  
    constructor(element) 
    { 
        this.element = element; 
        this.next = null
    } 
}
class LinkedList { 
    constructor() 
    { 
        this.head = null; 
        this.size = 0; 
    }
    add(element) 
{ 
    var node = new Node(element); 
  

    var current; 
  
    if (this.head == null) 
        this.head = node; 
    else { 
        current = this.head; 
   
        while (current.next) { 
            current = current.next; 
        }
        current.next = node; 
    } 
    this.size++;
} 

getLast() {
    let lastNode = this.head;
    if (lastNode) {
        while (lastNode.next) {
            lastNode = lastNode.next;
        }
    }
    return lastNode;
}
getFirst() {
    return this.head;
}

printList() 
{ 
    var curr = this.head; 
    var str = ""; 
    while (curr) { 
        str += curr.element + " "; 
        curr = curr.next; 
    } 
    console.log(str); 
} 
  
} 

my_linked_list = new LinkedList();
my_linked_list.add(12);
my_linked_list.add(34);
my_linked_list.add(23);
my_linked_list.add(400);
console.log(my_linked_list.getLast());
console.log(my_linked_list.getFirst())
my_linked_list.printList()

File: linear_data_structure/py/linked_list.py
class node:
    def __init__(self,data,link=None):
        self.data = data
        self.link = link

class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None

    def is_empty(self):
        if self.head == None:
            return True

    def append(self,value):
        new_node = node(value)
        if self.is_empty():
            self.head = new_node
            self.tail = new_node
        else:
            last_node = self.tail
            last_node.link = new_node
            self.tail = new_node
        print(value,"element inserted")

    def printlist(self):
        temp = self.head
        while temp!=None:
            print(temp.data)
            temp = temp.link

obj_linked_list = LinkedList()
obj_linked_list.append(50)
obj_linked_list.append(100)
obj_linked_list.append(150)

obj_linked_list.printlist()






